### PR TITLE
[3.x] Add hpc6id tests for develop

### PR DESCRIPTION
**Pull request by eantonin**
Signed-off-by: Edoardo Antonini <eantonin@amazon.com>

### Description of changes
* Add hpc6id tests for develop
* PR on 3.4 branch https://github.com/aws/aws-parallelcluster/pull/4658

### Tests
* Launched integ test in commercial Jenkins

### Checklist
- [X] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [X] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by francesco-giordano — DISMISSED** [review]

File: tests/integration-tests/tests/efa/test_efa.py [hunk 1]
```diff
@@ -197,8 +202,20 @@ def _test_osu_benchmarks_collective(
 
 
 def _test_osu_benchmarks_multiple_bandwidth(
-    remote_command_executor, scheduler_commands, test_datadir, slots_per_instance, partition=None
+    instance, remote_command_executor, scheduler_commands, test_datadir, slots_per_instance, partition=None
 ):
+    # Expected bandwidth with 4 NICS:
```
Comment (enrico-usai) [inline tests/integration-tests/tests/efa/test_efa.py]: these notes are valid for p4d instances only, let's highlight it or move these comments into the dict:

```python
instance_bandwidth_dict = {
        # Expected bandwidth for p4d and p4de (4 * 100 Gbps NICS -> declared NetworkPerformance 400 Gbps):
        # OMPI 4.1.0: ~330Gbps = 41250MB/s with Placement Group
        # OMPI 4.1.0: ~252Gbps = 31550MB/s without Placement Group
        # OMPI 4.0.5: ~95Gbps = 11875MB/s with Placement Group
        "p4d.24xlarge": 30000,  # Equivalent to a theoretical maximum of a single 240Gbps card
        "p4de.24xlarge": 30000,  # Equivalent to a theoretical maximum of a single 240Gbps card
        # # Expected bandwidth for hpc6id (2 up to 170 Gbps NICS -> declared NetworkPerformance 200 Gbps)
        "hpc6id.32xlarge": 23500,  # Equivalent to a theoretical maximum of a single 188Gbps card
    }
```

Comment (eantonin) [reply tests/integration-tests/tests/efa/test_efa.py]: good point, done

File: tests/integration-tests/tests/efa/test_efa.py [hunk 2]
```diff
@@ -197,8 +202,19 @@ def _test_osu_benchmarks_collective(
 
 
 def _test_osu_benchmarks_multiple_bandwidth(
-    remote_command_executor, scheduler_commands, test_datadir, slots_per_instance, partition=None
+    instance, remote_command_executor, scheduler_commands, test_datadir, slots_per_instance, partition=None
 ):
+    instance_bandwidth_dict = {
```
Comment (francesco-giordano) [inline tests/integration-tests/tests/efa/test_efa.py:207]: Minor not blocking, can you add the trn1.32xlarge performance?
8 NIC of 100 Gbps?

File: tests/integration-tests/tests/efa/test_efa.py [hunk 3]
```diff
@@ -215,11 +231,10 @@ def _test_osu_benchmarks_multiple_bandwidth(
         "cat /shared/osu_mbw_mr.out | tail -n +4 | awk '{print $2}' | sort -n | tail -n 1"
     ).stdout
 
-    # Expected bandwidth with 4 NICS:
-    # OMPI 4.1.0: ~330Gbps = 41250MB/s with Placement Group
-    # OMPI 4.1.0: ~252Gbps = 31550MB/s without Placement Group
-    # OMPI 4.0.5: ~95Gbps = 11875MB/s with Placement Group
-    expected_bandwidth = 30000
+    if instance_bandwidth_dict.get(instance) is None:
```
Comment (enrico-usai) [inline tests/integration-tests/tests/efa/test_efa.py]: minor:
```python

expected_bandwidth = instance_bandwidth_dict.get(instance)
if expected_bandwidth is None:
    pytest.fail(f"Instance {instance} is not valid for multiple bandwidth tests")
```

In this way you don't need to do the get another time.

Comment (eantonin) [reply tests/integration-tests/tests/efa/test_efa.py]: good point will fix

File: tests/integration-tests/tests/efa/test_efa.py (hunk 2 above)
Comment (eantonin) [reply tests/integration-tests/tests/efa/test_efa.py:207]: sure, done

**Review by francesco-giordano — APPROVED** [review]

**Pull request merged**